### PR TITLE
Use clap for command line argument parsing

### PR DESCRIPTION
**Pull request by felix-urban**
Using this library for command line parsing: https://docs.rs/clap/latest/clap/

Made it so we default to lnd's defaults if no arguments are passed and allow for much easier passing of the other arguments.

If we are going to do a real release it would probably make sense to remove the rust feature flags and just have those configurable.

Allows for nice help messages too:
image

### CODE REVIEW

File: src/main.rs [hunk 1]
```diff
@@ -189,37 +165,31 @@ async fn main() {
...
 
 #[cfg(feature = "redis")]
-fn parse_redis_config(mut args: std::env::ArgsOs) -> RedisStorage {
-    match args.next() {
-        Some(arg) => {
-            let str = arg.into_string().expect("Failed to parse redis config arg");
-            RedisStorage::new(str.as_str()).expect("Failed to create redis storage")
-        }
+fn parse_redis_config(cfg: Config) -> RedisStorage {
+    match cfg.db_path {
```
Comment (midnight-orange) [inline src/main.rs]:
```suggestion
    match cfg.redis_url {
```

File: src/main.rs [hunk 2]
```diff
@@ -189,37 +165,31 @@ async fn main() {
 }
 
 // name this _args to keep clippy happy when it's unused
-fn load_storage(mut _args: std::env::ArgsOs) -> Arc<Mutex<dyn Storage + Send>> {
+fn load_storage(_cfg: Config) -> Arc<Mutex<dyn Storage + Send>> {
```
Comment (midnight-orange) [inline src/main.rs]: maybe we just need to run clippy with both --sled and --redis flags

**Review by midnight-orange — CHANGES_REQUESTED** [review]

File: README.md [hunk 3]
```diff
@@ -11,20 +11,20 @@ HTLC interceptor to execute the theft.
...
 
 ```
-cargo run {LND_HOST} {LND_GRPC_PORT} {PATH_TO_LND_TLS_CERT} {PATH_TO_LND_ADMIN_MACAROON}
+cargo run -- --host {LND_HOST} --port {LND_GRPC_PORT} --cert-file {PATH_TO_LND_TLS_CERT} --macaroon-file {PATH_TO_LND_ADMIN_MACAROON}
 ```
 
 With sled db:
 ```
-cargo run --features sled {LND_HOST} {LND_GRPC_PORT} {PATH_TO_LND_TLS_CERT} {PATH_TO_LND_ADMIN_MACAROON}
+cargo run --features sled -- --host {LND_HOST} --port {LND_GRPC_PORT} --cert-file {PATH_TO_LND_TLS_CERT} --macaroon-file {PATH_TO_LND_ADMIN_MACAROON}
```
Comment (midnight-orange) [inline README.md]: nit: some of these sled / redis instructions might need some instructions for how to use the new db flag.

**Review by midnight-orange — APPROVED** [review]
readme instructions, otherwise lgtm

File: README.md (hunk 3 above)
Comment (felix-urban) [reply README.md]: Updated

**Pull request merged**